src/build/__init__.py

```diff
@@ -83,6 +89,11 @@ def check_version(requirement_string, extra=''):  # type: (str, str) -> bool
     for extra in req.extras:
         if extra not in (metadata.get_all('Provides-Extra') or []):
             return False
+        warnings.warn(
+            "Verified that the '{}[{}]' extra is present but did not verify that it is active "
```

Comment:
What does active mean in this context?

Reply:
Shouldn't we actually validate instead as that ticket alludes to?

Reply:
We verify that the distribution provides the extra but do not verify if the dependencies required to make the extra active are met. This will require us to recursively verify extras.

I wanted to a proper dependency checker into https://github.com/pypa/packaging and use it here. See https://github.com/pypa/packaging/issues/317.

For more details, look at the issue in the commit message.